kits/daisy/src/robot/hexapod.cpp

```diff
@@ -0,0 +1,610 @@
+      int num_legs_used = std::max((int)real_legs_.size(), 1);
+      int leg_count = 0;
+      for (int i = 0; i < 6 && leg_count < num_legs_used; ++i)
+      {
+        if (real_legs_.count(i) == 0)
+        {
+          continue;
+        }
+        leg_count++;
+        // Get from base of each leg
+        int num_prev_legs = std::count_if(real_legs_.begin(), real_legs_.end(), [i](int other_leg) { return other_leg < i; });
+        auto accel = fbk[num_prev_legs * num_leg_joints].imu().accelerometer().get(); 
+        auto gyro = fbk[num_prev_legs * num_leg_joints].imu().gyro().get(); 
+        Eigen::Vector3d mod_accel;
+        mod_accel << accel.getX(), accel.getY(), accel.getZ();
+        Eigen::Vector3d mod_gyro;
+        mod_gyro << gyro.getX(), gyro.getY(), gyro.getZ();
+        // Transform
+        Eigen::Matrix4d trans = legs_[i]->getKinematics().getBaseFrame();
+        avg_accel += trans.topLeftCorner(3,3) * mod_accel;
+        avg_gyro += trans.topLeftCorner(3,3) * mod_gyro;
+      }
+      // TODO: remove the 2.0!  This just seems to stabilize the motion right now, but probably because
+      // either:
+      // 1) there are other factors we don't have in there yet (e.g., grav comp)
+      // 2) the jacobian is slightly noisy
+      // Note that the foot forces have been compared to MATLAB results, and seem
+      // reasonable. No further debugging has been done.
+      avg_accel /= num_legs_used * 2.0;
+      avg_gyro /= num_legs_used * 2.0;
+
+      std::chrono::duration<double, std::ratio<1>> dt =
+        (std::chrono::steady_clock::now() - this->pose_start_time_);
+      this->pose_last_time_ = dt.count();
+    });
+    group->setFeedbackFrequencyHz(1000.0 / getFeedbackPeriodMs());
+  }
+}
+
+void Hexapod::startLogging()
+{
+  // Set up logging if enabled:
+  if (log_group_input_ || log_group_modules_)
+    std::cout << "Logging to 'logs' directory at " << params_.low_log_frequency_hz_ << "hz with bursts of " << params_.high_log_frequency_hz_ << " hz every 30 minutes." << std::endl;
+
+  std::string log_name_base;
+  {
+    char standard_file_name[40];
+    std::time_t now = std::time(nullptr);
+    std::tm* now_tm = std::localtime(&now);
+    std::snprintf(standard_file_name, 40, "log_file_%04d-%02d-%02d_%02d.%02d.%02d",
+                  1900 + now_tm->tm_year, 1 + now_tm->tm_mon, now_tm->tm_mday, now_tm->tm_hour,
+                  now_tm->tm_min, now_tm->tm_sec);
+    log_name_base = std::string(standard_file_name);
+  }
+
+  if (log_group_input_)
+  {
+    log_group_input_->setFeedbackFrequencyHz(params_.low_log_frequency_hz_);
+    log_group_input_->startLog("logs", log_name_base + "-IO.hebilog");
+  }
+  if (log_group_modules_)
+  {
+    log_group_modules_->setFeedbackFrequencyHz(params_.low_log_frequency_hz_);
+    log_group_modules_->startLog("logs", log_name_base + "-HEX.hebilog");
+
+  }
+}
+
+Hexapod::~Hexapod()
+{
+  if (group_)
+  {
+    group_->setFeedbackFrequencyHz(0);
```

Comment:
This is already done in their constructors?

Reply:
It didn't used to be :(

Reply:
Also, this is the destructor.

Reply:
I meant destructor, sorry